problems/arraysWithLoops.js

```diff
@@ -25,56 +40,117 @@ function shoutWhileLoop() {}
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
 
-function shoutForOfLoop() {}
+const shoutForOfLoop = (array) => {
+  let newArray = [];
+  for (let element of array) {
+    newArray.push(element + "!");
+  }
+  return newArray;
+};
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
 
-function sumArray() {}
+const sumArray = (nums) => {
+  let sum = 0;
+  for (let i = 0; i < nums.length; i++) {
+    sum += nums[i];
+  }
+  return sum;
+};
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
 
-function oddArray() {}
+const oddArray = (nums) => {
+  let newArray = [];
+  for (let i = 0; i < nums.length; i++) {
+    if (nums[i] % 2 === 1) {
+      newArray.push(nums[i]);
+    }
+  }
+  return newArray;
+};
 
 /**
  * Takes in an array and returns an array of all the even valued elements
  * @param {number[]} nums
  * @returns {array} evens
  */
 
-function evenArray() {}
+const evenArray = (nums) => {
+  let newArray = [];
+  for (let i = 0; i < nums.length; i++) {
+    if (nums[i] % 2 === 0) {
+      newArray.push(nums[i]);
+    }
+  }
+  return newArray;
+};
 
 /**
  * Takes in array and returns the smallest number in the array
  * @param {number[]} nums
  * @returns {number} smallest value
  */
 
-function findSmallest() {}
-
+const findSmallest = (nums) => {
+  let smallestNum = nums[0];
+  for (let i = 0; i < nums.length; i++) {
+    if( nums[i]< smallestNum) {
+      smallestNum = nums[i]
+    }
+  }
+  return smallestNum
+};
 /**
  * Takes in array and returns the second smallest number in the array
  * You may NOT use sort
  * @param {number[]} nums
  * @returns {number} second smallest value
  */
 
-function findSecondSmallest() {}
-
+const findSecondSmallest = (nums) => {
+  let smallestNum = Infinity
+  let secondSmallestNum = Infinity
+  for (let i = 0; i < nums.length; i++) {
+    if(nums[i] < smallestNum) {
+      secondSmallestNum = smallestNum;
+      smallestNum = nums [i]
+    } else if(nums [i] < secondSmallestNum) {
+      secondSmallestNum = nums[i]
+    }
+  } 
+  return secondSmallestNum
+      }
```

Comment:
Make sure you fix the placement of these curly braces so they line up with their correct scope